mdn/developer-portal
# 1643: Add "online" as location option

This changeset allows us to select "Online" as a location for an Event.

(Please read all the key changes, though)

Related issue #1643. 

Resolves #1177

## Key changes:

- Add support for "Online" as an option an event's "country" as part of the regular drop-down of countries
  - Note that this also makes "Online" an option for a Person too, but we don't have to use it / can just ignore it
- Instead of using the country code as display text in the card for an event and in the header strip for an event, we use the full/verbose name of the country. This is because the country code for the fake "Online" location is `QQ`, which makes no sense as a display string.
- Rephrase 'Country' in the user-facing site to Location (because that encompasses Online, too)
- Update phrasing in the CMS (non-public) to 'Country and Region' as requested in #1177
  - This could be further rephrased to just "Location", maybe?


![Screenshot 2020-07-03 at 10 36 09](https://user-images.githubusercontent.com/101457/86461153-76079400-bd21-11ea-8b30-1d7968e95407.png)
![Screenshot 2020-07-03 at 10 36 22](https://user-images.githubusercontent.com/101457/86461156-7738c100-bd21-11ea-8f82-0edefa1f337c.png)
![Screenshot 2020-07-03 at 10 42 31](https://user-images.githubusercontent.com/101457/86461157-7738c100-bd21-11ea-9a58-72efdeac906f.png)
![Screenshot 2020-07-03 at 10 42 37](https://user-images.githubusercontent.com/101457/86461159-77d15780-bd21-11ea-9719-2fd4fea63e38.png)
![Screenshot 2020-07-03 at 10 43 52](https://user-images.githubusercontent.com/101457/86461163-7869ee00-bd21-11ea-914e-651f126daa38.png)
![Screenshot 2020-07-03 at 10 43 57](https://user-images.githubusercontent.com/101457/86461165-79028480-bd21-11ea-82a9-e9630614c1d8.png)



## How to test

- Code is deployed to Stage
- Confirm that picking a country of "Online" is acceptable UX from an editor's POV
- Add/Edit an Event to have a location of "Online" 
  - confirm the card for the event looks ok / is appropriately labelled
  - confirm the header strip for that Event is also appropriate
  - confirm the location for other events is not broken/impaired
- Confirm filtering, including by location, still works as expected for:
  - Events
  - People
